crates/bevy_ui/src/flex/mod.rs

```diff
@@ -77,6 +77,7 @@ impl FlexSurface {
         style: &Style,
         calculated_size: CalculatedSize,
         scale_factor: f64,
+        preserve_aspect_ratio: bool,
```

Comment:
The logic of when we want to preserve the aspect ratio needs to be clearly documented in the code. Ideally in doc strings on this method.

The PR description is great, but not durable :)

Reply:
Yeah, I wasn't happy with the implementation, querying for images seemed dumb. Rewrote it when I woke up this morning so it adds a preserve aspect ratio flag to CalculatedSize instead, which seems so much better